File: App/app/src/main/java/com/cinema/ticket_booking/ui/booking/BookingDetailViewModel.java
package com.cinema.ticket_booking.ui.booking;

import androidx.lifecycle.*;
import com.cinema.ticket_booking.data.model.response.BookingResponse;
import com.cinema.ticket_booking.data.repository.BookingRepository;
import com.cinema.ticket_booking.util.Resource;
import dagger.hilt.android.lifecycle.HiltViewModel;
import javax.inject.Inject;

@HiltViewModel
public class BookingDetailViewModel extends ViewModel {
    private final BookingRepository repo;
    private final MutableLiveData<Resource<BookingResponse>> booking = new MutableLiveData<>();
    @Inject public BookingDetailViewModel(BookingRepository repo) { this.repo = repo; }
    public LiveData<Resource<BookingResponse>> getBooking() { return booking; }
    public void loadBooking(String id) { repo.getBookingDetail(id).observeForever(booking::setValue); }
}

File: App/app/src/main/java/com/cinema/ticket_booking/ui/booking/BookingDetailViewModel.kt
package com.cinema.ticket_booking.ui.booking

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.cinema.ticket_booking.data.model.response.BookingResponse
import com.cinema.ticket_booking.data.repository.BookingRepository
import com.cinema.ticket_booking.util.Resource
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class BookingDetailViewModel @Inject constructor(
    private val repo: BookingRepository
) : ViewModel() {

    private val _booking = MutableLiveData<Resource<BookingResponse>>()
    val booking: LiveData<Resource<BookingResponse>> = _booking

    fun loadBooking(id: String) {
        repo.getBookingDetail(id).observeForever { _booking.value = it }
    }

    fun cancelConfirm(token: String, bookingId: String): LiveData<Resource<Void>> {
        return repo.cancelConfirm(token, bookingId)
    }
}
